active/0002-packager.md

```diff
@@ -0,0 +1,257 @@
+- Start Date: 2015-03-01
+- RFC PR: (leave this empty)
+- ember-cli Issue: (leave this empty)
+
+# Summary
+
+When packaging Ember applications utilize the dependency graph to include only portions of the dep tree actually utilized. Additionally, hooks will be exposed to enable further innovation. HTTP2 packaging, custom bundles, multiple files etc.
+
+# Motivation
+
+The current app assembling strategy is naive as it merely merges and concats inputs trees in a nearly undocumented manner. Although this simplicity has served us well, it does limit flexibility of the build process. We believe we can further factor the build process in a way the improves the current À la carte strategy, while providing additional flexibitilty when desired.
+
+# Scope
+
+- Break the build up into 3 distinct phases 
+  - build
+  - satisfy
+  - package
+- Traverse a dependency graph to gain selective builds
+- Lay the infrastructure for the build-side of engines
+- deprecate the hard dependency on Bower in favor NPM
+- deprecate app.import API
+
+# Definitions
+
+- __Builder__ - Responsible for generating an array of broccoli trees that contain the app and all the addons.
+- __Pre-Packager__ - Responsible for resolving and reducing the graph to a tree containing only reachable modules.
+- __Packager__ - Responsible for taking the resolved tree and applying – default or user provided – concatenation strategies to the tree.
+- __Entry__ - A entry node into the graph in which you traverse down from. This can be an app or engine.
+
+# Detailed Design
+
+## Builder
```

Comment:
We should also mention that the builder will have the ability to consume both un-built and pre-built dependencies. This allows users to consume local branches of un-built add-ons like ember.js, and only then in-cure the initial build penalty. During the release process of add-ons, we pre-compute the `dep-graph.json` and the individual intermediate modules (named, but separate amd modules)